### PR TITLE
Added option to limit amount of lore lines

### PR DESCRIPTION
**Pull request by justinmtech**
### Information

This PR closes #4772

### Details

**Proposed feature:**  

Configure the max lines of lore that can be added with /itemlore add.
Adds config setting "max-item-lore" and an error message when a user tries to add a new line of lore to an item with the max.
https://github.com/EssentialsX/Essentials/discussions/4772

**Environments tested:**    

OS: Windows 10

Java version:  OpenJDK 17.0.1

- [x] Most recent Paper version (1.18.1, #153)
- [ ] CraftBukkit/Spigot/Paper 1.12.2
- [ ] CraftBukkit 1.8.8

**Demonstration:**  

Screenshot: [https://imgur.com/a/dQ6TedQ](url)
Plugin Download: [https://www.dropbox.com/s/jpl35j9rr57bs8m/EssentialsX-2.19.3-dev%2B10-b220c22-max-lore-feature-4772.jar?dl=0](url)

### CODE REVIEW

File: Essentials/src/main/resources/config.yml [hunk 1]
```diff
@@ -727,6 +727,9 @@ log-command-block-commands: true
 # Set the maximum speed for projectiles spawned with /fireball.
 max-projectile-speed: 8
 
+# Set the max lines of lore that can be added with /itemlore add
+max-item-lore: 10
```
Comment (JRoy) [inline Essentials/src/main/resources/config.yml]:
```suggestion
# Set the maximum amount of lore lines a user can set with the /itemlore command.
# Users with the essentials.itemlore.bypass permission will be able to bypass this limit.
max-itemlore-lines: 10
```

File: Essentials/src/main/java/com/earth2me/essentials/Settings.java [hunk 2]
```diff
@@ -1919,4 +1919,9 @@ public boolean isUpdateCheckEnabled() {
     public boolean showZeroBaltop() {
         return config.getBoolean("show-zero-baltop", true);
     }
+
+    @Override
+    public int getMaxItemLore() {
+        return config.getInt("max-item-lore", 10);
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/Settings.java]:
```suggestion
        return config.getInt("max-itemlore-lines", 10);
```

File: Essentials/src/main/resources/messages.properties [hunk 3]
```diff
@@ -554,6 +554,7 @@ itemloreCommandUsage2Description=Sets the specified line of the held item's lore
 itemloreCommandUsage3=/<command> clear
 itemloreCommandUsage3Description=Clears the held item's lore
 itemloreInvalidItem=\u00a74You need to hold an item to edit its lore.
+itemloreMaxLore=\u00a74You cannot add more lore to this item.
```
Comment (JRoy) [inline Essentials/src/main/resources/messages.properties]:
```suggestion
itemloreMaxLore=\u00a74You cannot add any more lore lines to this item.
```

**Review by JRoy — CHANGES_REQUESTED** [review]
Few things plus please add a `essentials.itemlore.bypass` permission to bypass this limit

**Comment by justinmtech** [comment]
Thanks! Looks good to me. I'll add the bypass shortly.

**Comment by mdcfe** [comment]
Hey @justinmtech, any updates on these changes?

**Review by JRoy — APPROVED** [review]

**Pull request merged**